src/main/java/org/candlepin/resteasy/filter/SecurityHoleAuthorizationFilter.java

```diff
@@ -102,18 +103,27 @@ void runFilter(ContainerRequestContext requestContext) {
             if (!securityHole.anonConsumer()) {
                 denyAccess(principal, method);
             }
-            AnonymousCloudConsumer anonymousCloudConsumer = anonymPrincipal.getAnonymousCloudConsumer();
-            String ownerKey = checkCloudAccountOrganization(anonymousCloudConsumer);
-            if (!ownerCurator.existsByKey(ownerKey)) {
-                throw new TooManyRequestsException(ORG_NOT_CREATED_IN_CANDLEPIN);
-            }
-            if (!poolCurator.hasPoolsForProducts(ownerKey, anonymousCloudConsumer.getProductIds())) {
-                throw new TooManyRequestsException(ORG_DOES_NOT_HAVE_POOLS);
+            validateAnonymousCloudConsumer(anonymPrincipal);
+        }
+        if (principal instanceof CloudConsumerPrincipal) {
+            if (!securityHole.anonConsumer()) {
```

Comment:
The name 'anonConsumer' in SecurityHole.java is not quite accurate now, since it is used to determine if either a standard autoreg token or an anonymous autoreg token user is authorized. Can we rename it into something like `autoregToken`?

Reply:
Done.